debugprofiler.md

```diff
@@ -0,0 +1,52 @@
+Forge Debug Profiler oraz jego interpretacja
+===============================================
+
+Spis treści
+-----------
+* [Podstawy](#podstawy)
+* [Jak używać debugera](#Jak_używać_debugera)
+* [Czytanie Profilera](#czytanie)
+
+## Podstawy
+Minecraft dostarcza nam `Debug Profiler` dzięki któremu można sprawdzić obciążający kod.
+Szczególnie chodzi tutaj o rzeczy pokroju `TickEvent` lub `Ticking TileEntities`, 
+przydaje się to w szcególności modderom, jak także właścicielom serwerów, którzy chcą sprawdzić przyczynę lagów.
+
+<a name="Jak_używać_debugera"><h2>Jak używać profilera</h2></a>
+
+Profiler jest banalnie prosty w obsłudze:
+- `/debug start` włącza nam Profiler (z konsoli, lub bezpośrednio w grze*)
+- `/debug stop` zatrzymuje nam Proflier (z konsoli, lub bezpośrednio w grze*)
+
+**Czym dłużej Proflier będzie włączony tym dokładniejsze dane ukaże. Zaleca się conajmniej minutę.**
+
+Ważne jest by podczas profilowania robić to co zawsze, chodzić po świecie etc. Można też przteleportować się w miejsce gdzie występują lagi. **Profiler nie zbiera danych które nie istnieją, tj. `Entity` jak i `TileEntities`, dlatego ważne jest by załadować chunki na których występują lagi**.
+
+Po zakończonym profilowaniu w folderze `debug/` zostanie utworzony plik o wzorze `profile-results-yyyy-mm-dd_hh.mi.ss.txt` na przykład: `profile-results-2019-12-28_22.52.14.txt`.
+
+<a name="czytanie"><h2>Czytanie Profilera</h2></a>
+
+Na samym początku napisane jest jak długo profiler działał (w milisekundach), i ile ticków w tym czasie było
+
+Poniżej znajduje się kod na wzór tego poniżej
```

Comment:
To jest bardziej lista lub graf niż kod.


Warto dodać, że im bardziej dana kategoria wysunięta jest w prawo, tym niżej w hierarchii się znajduję oraz podać przykład z wycinka poniżej: `levels`, czyli suma wszystkich poziomów nie są wcale wysunięte, a `minecraft:chest`, czyli pojedyncza skrzynia jest przesunięta aż o 5 pozycji (tabów) w prawo.